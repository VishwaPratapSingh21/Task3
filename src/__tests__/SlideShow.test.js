import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';
import Slideshow from '../components/SlideShow';

describe('Slideshow component', () => {
  const wrapper = shallow(<Slideshow />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders component in divs', () => {
    expect(wrapper.find('div')).toHaveLength(3);
  });

  it('renders component in span', () => {
    expect(wrapper.find('span')).toHaveLength(4);
  });

  it('renders component in li', () => {
    expect(wrapper.find('li')).toHaveLength(3);
  });

  it('renders component in ol', () => {
    expect(wrapper.find('ol')).toHaveLength(1);
  });
});
