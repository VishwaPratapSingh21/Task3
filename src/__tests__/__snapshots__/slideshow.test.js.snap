// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slideshow component renders properly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="carousel slide"
    data-ride="carousel"
    id="myCarousel"
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        data-slide-to="0"
        data-target="#myCarousel"
      />
      <li
        data-slide-to="1"
        data-target="#myCarousel"
      />
      <li
        data-slide-to="2"
        data-target="#myCarousel"
      />
    </ol>
    <div
      className="carousel-inner"
      role="listbox"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <FoodImages
        image="food4.jpg"
        status="active"
        text="family meal"
      />
      <FoodImages
        image="new1.jpg"
        status=""
        text="proteins one"
      />
      <FoodImages
        image="new2.jpg"
        status=""
        text="proteins"
      />
    </div>
    <a
      className="carousel-control-prev"
      data-slide="prev"
      href="#myCarousel"
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      data-slide="next"
      href="#myCarousel"
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>
</div>
`;
